_posts/2024-10-15-Announcing-Data-Prepper-2.10.0.md

```diff
@@ -0,0 +1,129 @@
+---
+layout: post
+title: Announcing Data Prepper 2.10.0
+authors:
+- dvenable
+- TODO
+date: 2024-10-15 12:30:00 -0600
+categories:
+  - releases
+excerpt: Data Prepper 2.10.0 offers an OpenSearch _bulk API and reads from Amazon Kinesis
+meta_keywords: Data Prepper, _bulk, Amazon Kinesis Data Streams, OTel Logs, OTLP JSON
+meta_description: Data Prepper 2.10.0 offers a source that simulates the OpenSearch _bulk API and another source for reading from Amazon Kinesis Data Streams.
+---
+
+## Introduction
+
+Data Prepper 2.10 is now available for the community to use. 
+Two major features include a source to send data to Data Prepper using an API mimicking the OpenSearch `_bulk` API from OpenSearch and reading from Amazon Kinesis Data Streams.
+
+
+## OpenSearch API source
+
+Many existing OpenSearch clients that perform ingestion directly to OpenSearch can now send that data to Data Prepper first.
+With this, you can use Data Prepper's buffering and rich processor set before sending data to OpenSearch without having to change clients when they are using the OpenSearch `_bulk` API. 
+A new source has been added in Data Prepper named `opensearch_api` that accepts [OpenSearch Document API Bulk operation](https://opensearch.org/docs/latest/api-reference/document-apis/bulk/) requests from clients using REST and ingests data into OpenSearch. 
+The behavior of this source is also quite similar to the existing `http` source. 
+It supports industry-standard encryption in the form of TLS/HTTPS and HTTP basic authentication. 
+It parses incoming requests and create Data Prepper events and associated event metadata making it compatible with the `opensearch` sink. 
+The request body should be compatible with OpenSearch Document API Bulk Operation and will also support all the actions like index, create, delete and update.
+
+The two HTTP methods supported now are the following:
+
+```
+POST _bulk
+POST <index>/_bulk
+```
+
+The second API which specifies the index in the path means you don’t need to include it in the request body.
+
+Moreover, the following query parameters are also supported that are available in OpenSearch Document API Bulk Operation as the following below:
+
+* pipeline
```

Comment:
Should these both be capitalized?

Reply:
No, these are keywords. I'll use backticks to clarify.